### PR TITLE
docs(HostingReactApp): enables amplify hosting option for example app

### PR DESCRIPTION
**Pull request by TheEvilDev**
## Overview
This change sets up the example React app to be "end to end" instead of just client side. By doing an amplify push to your AWS account, the example app will create the necessary resources (eventually) to run all the main elements of the demo.

So far in this change:
- Hosting enabled
- Basic Auth (username/password, sign up flow)
- Auth Role permissions extended to enable talking to TwinMaker and KVS
- AWS TimeStream database and table setup
- Partial REST API implemented that will serve as endpoints to trigger mock data generation
- Removed .env.local settings, and changed react components so credentials are recieved from signing in instead of local env
- Created SignIn/SignUp Page

## Verifying Changes
Current state can be found here: https://dev.d29vgcf5diodtm.amplifyapp.com/ (just sign up for an account).

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: examples/react-app/src/hooks/useUserInfo.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          raw: currentUserInfo,
+        };
+
+        setUserInfo(userInfo);
+      } else {
+        navigate('/signin')
+      }
+    };
+
+    doAsync();
```
Comment (hartmir) [inline examples/react-app/src/hooks/useUserInfo.ts:40]: Is there any benefit to converting to a pure function? That is, return the creds in a Promise, then `doAsync().then(setCredentials).catch(navigate('/signin'));`

File: examples/react-app/src/hooks/useTwinMakerDatasource.ts
Comment (hartmir) [inline examples/react-app/src/hooks/useTwinMakerDatasource.ts:1]: This is much more user-friendly. Do you have a plan to update generated docs with the new hook?

**Review by hartmir — CHANGES_REQUESTED** [review]
Great change, exactly what I was struggling to do myself.

File: examples/react-app/src/components/VideoPlayer.tsx [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
-      <VideoPlayer videoData={videoData} viewport={videoViewport} />
-    </div>
-  );
+const VideoPlayerComponent: FC<VideoPlayerComponentProps> = ({ workspace }) => {
+  const datasource = useTwinMakerDatasource(workspace);
+  const videoData = datasource?.videoData({
+    kvsStreamName: kvsStreamName,
+    entityId: videoEntityId,
+    componentName: videoComponentName,
+  });
```
Comment (hartmir) [inline examples/react-app/src/components/VideoPlayer.tsx:16]: Consider wrapping in `useMemo` as in SceneViewer

**Comment by diehbria** [comment]
closing as stale PR

**Pull request closed without merging**